src/client/consumer.ts

```diff
@@ -70,29 +74,35 @@ class Consumer implements vscode.Disposable {
      * Received messages and/or errors are emitted via events.
      */
     async start(): Promise<void> {
+        const partition = this.options.partition;
+        const fromOffset = this.options.fromOffset;
+        const topic = this.options.topicId;
         this.kafkaClient = createKafka(this.options);
-        this.consumer = this.kafkaClient.consumer({ groupId: `vscode-kafka-${this.options.clusterId}-${this.options.topic}`, retry: { retries: 3 }});
+        this.consumer = this.kafkaClient.consumer({ groupId: this.options.consumerGroupId, retry: { retries: 3 } });
         await this.consumer.connect();
-        await this.consumer.subscribe({ topic: this.options.topic, fromBeginning: this.options.fromOffset ===  "earliest" })
 
-        await this.consumer.run({
-            eachMessage: async ({ topic , partition, message }) => {
+        const subscribeOptions = this.createSubscribeOptions(topic, fromOffset);
+        await this.consumer.subscribe(subscribeOptions)
+
+        this.consumer.run({
+            eachMessage: async ({ topic, partition, message }) => {
                 this.onDidReceiveMessageEmitter.fire({
                     uri: this.uri,
                     record: { topic: topic, partition: partition, ...message },
                 });
             },
         });
-    }
 
-    private parseUri(uri: vscode.Uri): { clusterId: string; topic: string; partition?: string} {
-        const [clusterId, topic, partition] = uri.path.split("/");
+        if (partition || (fromOffset && subscribeOptions.fromBeginning == undefined)) {
```

Comment:
```typescript
partition && (fromOffset && subscribeOptions.fromBeginning == undefined)
```

Else, you'll run into errors like :
```
stack trace: KafkaJSNonRetriableError: Invalid offset, expected a long received latest
```
with a consumer like:
```
CONSUMER foo
topic: bar
partition:1
```

Reply:
actually the partition || fromOffset logic is correct, but needs to check whether fromOffset is a valid integer

Reply:
ok I did validation for offset and partition in the start/stop command handler.